lib/ConvoyLeader/src/App.cpp

```diff
@@ -326,6 +319,81 @@ void App::fatalErrorHandler()
     }
 }
 
+bool App::setupMqttClient()
+{
+    /* Setup MQTT Server, Birth and Will messages. */
+    bool                                           isSuccessful = false;
+    SettingsHandler&                               settings     = SettingsHandler::getInstance();
+    StaticJsonDocument<JSON_BIRTHMESSAGE_MAX_SIZE> birthDoc;
+    char                                           birthMsgArray[JSON_BIRTHMESSAGE_MAX_SIZE];
+    String                                         birthMessage;
+
+    birthDoc["name"] = settings.getRobotName();
+    (void)serializeJson(birthDoc, birthMsgArray);
+    birthMessage = birthMsgArray;

```

Comment:
Hmm ... having already the birth message in an array and allocating it additional on the heap ... could we avoid it by changing the MqttSettings structure?